### PR TITLE
chore: Ensure that che-server's SA have enough permission to manipulate with devworkspaces resource on kubernetes

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### What does this PR do?
che-operator delegates some permissions to manipulates with devworkspace resources

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/19920

### How to test this PR?

1. deploy Eclipse Che
2. check if `eclipse-che-cheworkspaces-devworkspace-clusterrole` cluster role & cluster role binding are created

Image to test: `docker.io/abazko/operator:19920`

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [Nightly OLM bundle is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-nightly-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by mmorhun — APPROVED** [review]

File: pkg/controller/che/workspace_namespace_permission.go [hunk 1]
```diff
@@ -139,19 +147,72 @@ func (r *ReconcileChe) removeNamespaceEditorPermissions(deployContext *deploy.De
...
 	}
 
 	return true, nil
 }
 
+func getDevWorkspacePolicies() []rbac.PolicyRule {
+	k8sPolicies := []rbac.PolicyRule{
+		{
+			APIGroups: []string{"workspace.devfile.io"},
+			Resources: []string{"devworkspaces", "devworkspacetemplate"},
```
Comment (skabashnyuk) [inline pkg/controller/che/workspace_namespace_permission.go:208]: devworkspacetemplate -> devworkspacetemplates

File: pkg/controller/che/workspace_namespace_permission_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		t.Fatalf("Finalizer '%s' not added", DevWorkspacePermissionsFinalizerName)
+	}
+
+	name := "eclipse-che-cheworkspaces-clusterrole"
+	exists, _ := deploy.Get(deployContext, types.NamespacedName{Name: name}, &rbac.ClusterRole{})
+	if !exists {
+		t.Fatalf("Cluster Role '%s' not found", name)
+	}
+
+	name = "eclipse-che-cheworkspaces-clusterrole"
```
Comment (AndrienkoAleksandr) [inline pkg/controller/che/workspace_namespace_permission_test.go:58]:
```suggestion
```
clusterrole and clusterrolebinding name are the same.

File: pkg/controller/che/workspace_namespace_permission_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		t.Fatalf("Cluster Role Binding '%s' not found", name)
+	}
+
+	name = "eclipse-che-cheworkspaces-namespaces-clusterrole"
+	exists, _ = deploy.Get(deployContext, types.NamespacedName{Name: name}, &rbac.ClusterRole{})
+	if !exists {
+		t.Fatalf("Cluster Role '%s' not found", name)
+	}
+
+	name = "eclipse-che-cheworkspaces-namespaces-clusterrole"
```
Comment (AndrienkoAleksandr) [inline pkg/controller/che/workspace_namespace_permission_test.go:70]:
```suggestion
```

File: pkg/controller/che/workspace_namespace_permission_test.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		t.Fatalf("Cluster Role Binding '%s' not found", name)
+	}
+
+	name = "eclipse-che-cheworkspaces-devworkspace-clusterrole"
+	exists, _ = deploy.Get(deployContext, types.NamespacedName{Name: name}, &rbac.ClusterRole{})
+	if !exists {
+		t.Fatalf("Cluster Role '%s' not found", name)
+	}
+
+	name = "eclipse-che-cheworkspaces-devworkspace-clusterrole"
```
Comment (AndrienkoAleksandr) [inline pkg/controller/che/workspace_namespace_permission_test.go:82]:
```suggestion
```

**Pull request merged**